File: Longest Substring Without Repeating Characters.java
import java.util.HashSet;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length() == 0) {
            return 0;
        } else {
            char[] s1 = s.toCharArray();
            int result = 0;
            int count = 0;
            HashSet<Character> set = new HashSet<>();
            for (int i = 0; i < s1.length; i++) {
                if (set.contains(s1[i])) {
                    while (set.contains(s1[i])) {
                        set.remove(s1[i - count]);
                        count--;
                    }
                }
                count++;
                set.add(s1[i]);
                if (count > result) {
                    result = count;
                }
            }
            return result;
        }
    }
}

File: Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        arr=list(s)
        for i in range(len(arr)):
            c = 0
            d = {}
            a = arr[i]
            j = i
            while (a not in d and j < len(arr)):
               
               d[a] = 1
               c+=1
               j+=1
               if(j < len(arr)):
                a = arr[j]
               
            if c > count:
                count = c
        return count
